icu4c/source/test/intltest/strcase.cpp

```diff
@@ -1330,7 +1331,8 @@ void StringCaseTest::TestCaseMapUTF8WithEdits() {
 
     edits.reset();
     length = CaseMap::utf8ToUpper("el", U_OMIT_UNCHANGED_TEXT,
-                                  reinterpret_cast<const char*>(u8"Πατάτα"), 6 * 2, dest, UPRV_LENGTHOF(dest), &edits, errorCode);
+                                  reinterpret_cast<const char*>(u8"Πατάτα"), 6 * 2,
```

Comment:
I suspect this `reinterpret_cast` is no longer needed (such uses were removed in similar tests)

Reply:
On further reflection, I think I was wrong here; the invoked overload is one that has a corresponding `const char*` parameter type.  The other tests where `reinterpret_cast` was removed correspond to cases that use `StringPiece`.

Reply:
Right. We have two versions of these functions, taking pointer+length vs. StringPiece. These unit tests are both for functionality and for API coverage.